requests/request_body.go

```diff
@@ -29,9 +30,20 @@ type RequestBodyValidator interface {
 	ValidateRequestBodyWithPathItem(request *http.Request, pathItem *v3.PathItem, pathValue string) (bool, []*errors.ValidationError)
 }
 
+type Config struct {
+	RegexEngine jsonschema.RegexpEngine
+}
+
 // NewRequestBodyValidator will create a new RequestBodyValidator from an OpenAPI 3+ document
-func NewRequestBodyValidator(document *v3.Document) RequestBodyValidator {
-	return &requestBodyValidator{document: document, schemaCache: &sync.Map{}}
+func NewRequestBodyValidator(document *v3.Document, cfg ...Config) RequestBodyValidator {
+
+	config := Config{} // Default
+
+	if len(cfg) > 0 {
+		config = cfg[0]
+	}
+
+	return &requestBodyValidator{Config: config, document: document, schemaCache: &sync.Map{}}
 }
```

Comment:
Here and everywhere

I disagree with such pattern.

Using a variadic for passing one optional value, and only one is not something good.


I would like you to suggest the Option pattern

```suggestion
type ConfigFunc func(*Config)

func WithRegexEngine(engine jsonschema.RegexpEngine) ConfigFunc {
    return func(c *Config) {
        c.RexgepEngine = engine
    }
}

func NewRequestBodyValidator(document *v3.Document, cfgs ...ConfigFunc) RequestBodyValidator {

	config := Config{} // Default

	for _, cfg := range cfgs {
		opt(&config)
	}

	return &requestBodyValidator{Config: config, document: document, schemaCache: &sync.Map{}}
 }
```

This could be extended to use other overloads

So if you have a need for overloading the whole Config

You can add something like this

```go
func WithConfig(newConf Config) ConfigFunc {
    return func(c *Config) {
        *c = &newConf
    }
}
```

This is pseudo code, I'm writting on a phone

Reply:
Makes perfect sense ... will make those changes..